importlib_metadata/__init__.py

```diff
@@ -831,7 +831,13 @@ def __init__(self, path: FastPath):
         self.infos = FreezableDefaultDict(list)
         self.eggs = FreezableDefaultDict(list)
 
-        for child in path.children():
+        for child in sorted(path.children()):
+            # Empty directories aren't interesting
+            childpath = path.joinpath(child)
+            with suppress(PermissionError):
+                if childpath.is_dir() and not any(childpath.iterdir()):
```

Comment:
I'm fairly confident this check is going to slow down the behavior. I think it also changes the behavior, causing metadata directories not to be recognized.